postproengine/reynoldsstress.py

```diff
@@ -123,11 +142,74 @@ def execute(self, verbose=False):
                     actionitem.execute()
         return 
 
+
+    @registeraction(actionlist)
+    class compute_radial_shear_stress():
+        actionname = 'radial_stress'
+        blurb      = 'Computes the radial Reynolds shear stress from the Cartesian stresses'
+        required   = False
+        actiondefs = [
+                {'key':'iplane', 'required':False, 'default':None, 'help':'List of iplane values. Default is all planes in ncfile.'},
+                {'key':'yc', 'required':False, 'default':0, 'help':'Specified lateral center of wake, yc'},
+                {'key':'zc', 'required':False, 'default':0, 'help':'Specified vertical center of wake, zc'},
+                {'key':'wake_center_files', 'required':False, 'default':None, 'help':'csv files containing time series of wake centers for each iplane. yc and zc will be compute based on mean centers over the specified time interval'},
+        ]
+
+        def __init__(self, parent, inputs):
+            self.actiondict = mergedicts(inputs, self.actiondefs)
+            self.parent = parent
+            print('Initialized '+self.actionname+' inside '+parent.name)
+            return
+
+        def execute(self):
+            yc = self.actiondict['yc']
+            zc = self.actiondict['zc']
+            iplanes = self.actiondict['iplane']
+            wake_center_files = self.actiondict['wake_center_files']
+
+            if iplanes == None:
+                iplanes = list(range(0,self.parent.dbReAvg['x'].shape[0]))
+
+            if not isinstance(iplanes, list): iplanes = [iplanes,]
+            if not wake_center_files == None and not isinstance(wake_center_files, list): wake_center_files = [wake_center_files,]
+
+            if wake_center_files != None and len(wake_center_files) != len(iplanes):
+                print("Error: len(wake_center_files) != len(iplanes). Exiting.")
+                sys.exit()
+
+            self.parent.dbReAvg['uxur_avg']    = np.zeros_like(self.parent.dbReAvg['uv_avg'])
+            self.parent.dbReAvg['ux_avg_uxur_avg'] = np.zeros_like(self.parent.dbReAvg['uv_avg'])
+            for iplaneiter, iplane in enumerate(iplanes):
+                if wake_center_files != None:
+                    df = pd.read_csv(wake_center_files[iplaneiter])
+                    filtered_df = df[(df['t'] >= self.parent.tavg[0]) & (df['t'] <= self.parent.tavg[-1])]
+                    y_center = filtered_df['yc'].mean()
+                    z_center = filtered_df['zc'].mean()
+                else:
+                    y_center = yc
+                    z_center = zc
+
+                YY = self.parent.dbReAvg['y'][iplane,:,:]
+                ZZ = self.parent.dbReAvg['z'][iplane,:,:]
+                uv_avg  = self.parent.dbReAvg['uv_avg'][iplane,:,:]
+                uw_avg  = self.parent.dbReAvg['uw_avg'][iplane,:,:]
+                ux_avg  = self.parent.dbReAvg['velocityx_avg'][iplane,:,:]
+
+                Theta = np.arctan2(ZZ-z_center,YY-y_center)
```

Comment:
Can you double check this part? Should match equation (22) in this reference: https://www.tandfonline.com/doi/abs/10.1080/14685248.2012.705005

Reply:
This looks correct.